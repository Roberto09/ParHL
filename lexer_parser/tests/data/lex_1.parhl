let x : float := 1 + 3.2 * 8 % 10 ^ 5, y : string := "hola";
print(x)
print(y)
if(True){
    x := 10
}

if(False)
{
    a := read_line()
} else {
    b := read_file("test.txt")
}

if(a=b) {

} else if(a > b) {
    write_file()
}


while(a > 0) {
    b:= a or d;
}

for(let a : int := 0; a < 10; a:= a+1) {
    y:= y * a;
}

let foo(a : string ) : int {
    print("hello")
    return 2
} 

foo("name")