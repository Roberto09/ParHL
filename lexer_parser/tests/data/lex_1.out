Token(type='LET', value='let', lineno=1, index=0)
Token(type='ID', value='x', lineno=1, index=4)
Token(type='COLON', value=':', lineno=1, index=6)
Token(type='FLOAT_T', value='float', lineno=1, index=8)
Token(type='ASSIG', value=':=', lineno=1, index=14)
Token(type='INT_V', value=1, lineno=1, index=17)
Token(type='PLUS', value='+', lineno=1, index=19)
Token(type='FLOAT_V', value=3.2, lineno=1, index=21)
Token(type='MULT', value='*', lineno=1, index=25)
Token(type='INT_V', value=8, lineno=1, index=27)
Token(type='MOD', value='%', lineno=1, index=29)
Token(type='INT_V', value=10, lineno=1, index=31)
Token(type='EXP', value='^', lineno=1, index=34)
Token(type='INT_V', value=5, lineno=1, index=36)
Token(type='COMMA', value=',', lineno=1, index=37)
Token(type='ID', value='y', lineno=1, index=39)
Token(type='COLON', value=':', lineno=1, index=41)
Token(type='STRING_T', value='string', lineno=1, index=43)
Token(type='ASSIG', value=':=', lineno=1, index=50)
Token(type='STRING_V', value='hola', lineno=1, index=53)
Token(type='SEMICOLON', value=';', lineno=1, index=59)
Token(type='NEWLINE', value='\n', lineno=1, index=60)
Token(type='PRINT', value='print', lineno=1, index=61)
Token(type='L_PAREN', value='(', lineno=1, index=66)
Token(type='ID', value='x', lineno=1, index=67)
Token(type='R_PAREN', value=')', lineno=1, index=68)
Token(type='NEWLINE', value='\n', lineno=1, index=69)
Token(type='PRINT', value='print', lineno=1, index=70)
Token(type='L_PAREN', value='(', lineno=1, index=75)
Token(type='ID', value='y', lineno=1, index=76)
Token(type='R_PAREN', value=')', lineno=1, index=77)
Token(type='NEWLINE', value='\n', lineno=1, index=78)
Token(type='IF', value='if', lineno=1, index=79)
Token(type='L_PAREN', value='(', lineno=1, index=81)
Token(type='BOOL_V', value=True, lineno=1, index=82)
Token(type='R_PAREN', value=')', lineno=1, index=86)
Token(type='L_BRACE', value='{', lineno=1, index=87)
Token(type='NEWLINE', value='\n', lineno=1, index=88)
Token(type='ID', value='x', lineno=1, index=93)
Token(type='ASSIG', value=':=', lineno=1, index=95)
Token(type='INT_V', value=10, lineno=1, index=98)
Token(type='NEWLINE', value='\n', lineno=1, index=100)
Token(type='R_BRACE', value='}', lineno=1, index=101)
Token(type='NEWLINE', value='\n', lineno=1, index=102)
Token(type='ELSE', value='else', lineno=1, index=103)
Token(type='L_BRACE', value='{', lineno=1, index=107)
Token(type='NEWLINE', value='\n', lineno=1, index=108)
Token(type='ID', value='x', lineno=1, index=113)
Token(type='ASSIG', value=':=', lineno=1, index=115)
Token(type='INT_V', value=20, lineno=1, index=118)
Token(type='SEMICOLON', value=';', lineno=1, index=120)
Token(type='NEWLINE', value='\n', lineno=1, index=121)
Token(type='R_BRACE', value='}', lineno=1, index=122)
